### PR TITLE
Include filters when exporting dashboard to PDF

### LINKED ISSUE #35475: Export to PDF should have the filter values that were used on the dashboard
**Is your feature request related to a problem? Please describe.**
A request that came up twice on the same week: print in the PDF also the values that were used when the PDF was generated

**Describe the solution you'd like**
Just add the filter and the values at the top

**Describe alternatives you've considered**
NA

**How important is this feature to you?**
Requested 2 times on the same week

**Additional context**
NA

### PR DESCRIPTION
**Pull request by perivamsi**
Closes https://github.com/metabase/metabase/issues/35475

### Demos

<details>
<summary>No filters</summary>

CleanShot 2024-02-22 at 14 04 42@2x

</details>

<details>
<summary>With filters, no values</summary>

CleanShot 2024-02-22 at 14 06 16@2x

</details>

<details>
<summary>With filters and values</summary>

CleanShot 2024-02-22 at 14 08 39@2x

</details>

<details>
<summary>Full width dashboard</summary>

              
CleanShot 2024-02-22 at 14 09 39@2x

</details>

### CODE REVIEW

**Comment by perivamsi** [comment]
PDF without this change
Screenshot 2024-01-29 at 11 36 30 AM

PDF with this change (and no filter values selected)
Screenshot 2024-01-29 at 11 36 41 AM


PDF with this change (and with filter values selected)
Screenshot 2024-01-29 at 11 36 50 AM

**Comment by perivamsi** [comment]
The filter labels look funky, they overlap with the filter control.

**Review by EmmadUsmani — CHANGES_REQUESTED** [review]
I think there's a couple of things that still need to be fixed visually
- The padding to the left of the title is missing
- Dashcard outlines are no longer visible

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.tsx [hunk 1]
```diff
@@ -123,11 +113,34 @@ export const ParametersWidgetContainer = styled(FullWidthContainer)<{
...
+    ${ParametersWidgetContainer} {
+      background-color: transparent;
+      border-bottom: none;
+
+      margin-top: 1rem;
+
+      legend {
+        top: -12px;
+      }
+    }
```
Comment (kulyk) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.tsx:139]: Workaround: `html2canvas` doesn't handle `fieldset` + `legend` elements we're using for filters
https://github.com/niklasvh/html2canvas/issues/2418

cleanshot_2024-02-21_at_17 03 30_2x

**Review by cdeweyx — APPROVED** [review]

**Review by alxnddr — APPROVED** [review]

File: frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.tsx [hunk 2]
```diff
@@ -641,7 +641,7 @@ class DashboardHeaderContainer extends Component<DashboardHeaderProps> {
 
   saveAsPDF = async () => {
     const { dashboard } = this.props;
-    const cardNodeSelector = "#Dashboard-Cards-Container";
+    const cardNodeSelector = "#Dashboard-Parameters-And-Cards-Container";
```
Comment (EmmadUsmani) [inline frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.tsx:644]: Nit: extracting this to a single constant that we share between this component and `Dashboard.tsx` would be nice

**Review by EmmadUsmani — APPROVED** [review]

**Comment by kulyk** [comment]
@metabase-bot backport release-x.49.x

**Comment by kulyk** [comment]
@metabase-bot backport release-x.48.x

**Pull request merged**